packages/flutter_rx_bloc/lib/src/base/stream_multi_builder_3.dart

```diff
@@ -0,0 +1,269 @@
+
+  Widget build(
+    BuildContext context,
+    S1 currentSummary1,
+    S2 currentSummary2,
+    S3 currentSummary3,
+  );
+
+  @override
+  State<_StreamMultiBuilderBase3<T1, T2, T3, S1, S2, S3>> createState() =>
+      _StreamMultiBuilderBase3State<T1, T2, T3, S1, S2, S3>();
+}
+
+class _StreamMultiBuilderBase3State<T1, T2, T3, S1, S2, S3>
+    extends State<_StreamMultiBuilderBase3<T1, T2, T3, S1, S2, S3>> {
+  final _subscriptionContainer1 = _StreamSubscriptionContainer<T1, S1>();
+  final _subscriptionContainer2 = _StreamSubscriptionContainer<T2, S2>();
+  final _subscriptionContainer3 = _StreamSubscriptionContainer<T3, S3>();
+
+  @override
+  void initState() {
+    super.initState();
+
+    // Setup initial state
+    _subscriptionContainer1.summary = widget.streamContainer1.initial();
+    _subscriptionContainer2.summary = widget.streamContainer2.initial();
+    _subscriptionContainer3.summary = widget.streamContainer3.initial();
+
+    // Setup subscriptions
+    _subscriptionContainer1.subscribe(widget.streamContainer1, setState);
+    _subscriptionContainer2.subscribe(widget.streamContainer2, setState);
+    _subscriptionContainer3.subscribe(widget.streamContainer3, setState);
+  }
+
+  @override
+  void didUpdateWidget(
+      _StreamMultiBuilderBase3<T1, T2, T3, S1, S2, S3> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    // Update subscriptions
+    _subscriptionContainer1.update(
+        oldWidget.streamContainer1.stream, widget.streamContainer1, setState);
+    _subscriptionContainer2.update(
+        oldWidget.streamContainer2.stream, widget.streamContainer2, setState);
+    _subscriptionContainer3.update(
+        oldWidget.streamContainer3.stream, widget.streamContainer3, setState);
+  }
+
+  @override
+  Widget build(BuildContext context) => widget.build(
+        context,
+        _subscriptionContainer1.summary,
+        _subscriptionContainer2.summary,
+        _subscriptionContainer3.summary,
+      );
+
+  @override
+  void dispose() {
+    _subscriptionContainer1.unsubscribe();
+    _subscriptionContainer2.unsubscribe();
+    _subscriptionContainer3.unsubscribe();
+    super.dispose();
+  }
+}
+
+/// endregion
+
+/// region _StreamContainer
+
+class _StreamContainer<T> extends _StreamContainerBase<T, AsyncSnapshot<T>> {
+  _StreamContainer({
+    required super.stream,
+    this.initialData,
+  });
+
+  final T? initialData;
+
+  @override
+  AsyncSnapshot<T> initial() => initialData == null
+      ? AsyncSnapshot<T>.nothing()
+      : AsyncSnapshot<T>.withData(ConnectionState.none, initialData as T);
+
+  @override
+  AsyncSnapshot<T> afterConnected(AsyncSnapshot<T> current) =>
+      current.inState(ConnectionState.waiting);
+
+  @override
+  AsyncSnapshot<T> afterData(AsyncSnapshot<T> current, T data) {
+    return AsyncSnapshot<T>.withData(ConnectionState.active, data);
+  }
+
+  @override
+  AsyncSnapshot<T> afterError(
+      AsyncSnapshot<T> current, Object error, StackTrace stackTrace) {
```

Comment:
This could be changed to use expression body `=>` instead of return
https://dart.dev/tools/linter-rules/prefer_expression_function_bodies and this lint could probably be added to analysis_options.yaml